### PR TITLE
72219 Application call stack for messages 

### PR DESCRIPTION
**Pull request by agustinvillar**
First approach to appStack solution

===CODE REVIEW===
- [ ] @drewburlingame 

===Additional Requirements===
 - [ ] [Feature is dev complete](https://daptiv.tpondemand.com/entity/72219)

### CODE REVIEW

File: Druq.Hub/Messaging/EnvelopeProcessor.cs [hunk 1]
```diff
@@ -4,14 +4,18 @@
...
 using ObjectPrinter;
 using ServiceStack.Text;
 
+
 namespace Druq.Hub.Messaging
 {
+    using Daptiv.Bcl.Diagnostics;
+    using IMetricsLog = Druq.Framework.Logs.IMetricsLog;
+    using MetricsLogManager = Druq.Framework.Logs.MetricsLogManager;
+
```
Comment (drewburlingame) [inline Druq.Hub/Messaging/EnvelopeProcessor.cs]: you could replace these 3 usings with using AppStack=Daptiv.Bcl.Diagnostics.AppStack

File: Druq.Hub/Messaging/EnvelopeProcessor.cs [hunk 2]
```diff
@@ -242,7 +245,10 @@ public void ProcessEnvelope(Envelope envelope)
         
         private void ProcessMessage(SubscriberMetadata subscriberMetadata, object subscriberInstance, object message)
         {
-            subscriberMetadata.ReceiveMethod.Invoke(subscriberInstance, new[] {message});
+            using(AppStack.Push(subscriberInstance.GetType().Name))
```
Comment (drewburlingame) [inline Druq.Hub/Messaging/EnvelopeProcessor.cs]: yes, this looks like the right place to me.

**Comment by agustinvillar** [comment]
@drewburlingame, here the Bcl should be managed by nugget? I did not find any place to change version to latest (1.0.42)

**Comment by agustinvillar** [comment]
Daptiv.Bcl is not accessible outside of Daptiv, so all the mapping will be performed within PPM

**Pull request closed without merging**